[🐛] [Android] Build fails in bare Expo project when integrating react-native-google-mobile-ads with @react-native-firebase/analytics

### What happened?

### Description:
When integrating both react-native-google-mobile-ads and @react-native-firebase/analytics (with @react-native-firebase/app) in a bare Expo project, the Android build fails due to a conflict in the AndroidManifest.xml.

The conflict arises because both libraries attempt to declare the same <property> attribute (android.adservices.AD_SERVICES_CONFIG) in their respective manifests (play-services-ads-lite and play-services-measurement-api). To resolve the issue, a manual addition of tools:replace="android:resource" is required in the <property> element of the AndroidManifest.xml.

1. Create a new bare Expo project:
```
$ expo init
Choose the minimal and bare template.
```

2. Install the required libraries:
```
yarn add react-native-google-mobile-ads
yarn add @react-native-firebase/analytics
yarn add @react-native-firebase/app
yarn add expo-build-properties
```

3. Place google-services.json in the root directory of the project.

4. Attempt to build the Android project:

`npx expo run:android`

Expected Behavior:
The project should build successfully without requiring manual changes to AndroidManifest.xml.

### ErrorLogs:

```
$npx expo run:android

› Building app...
Configuration on demand is an incubating feature.

> Configure project :react-native-firebase_analytics
:react-native-firebase_analytics package.json found at /Users/ken/my-app/node_modules/@react-native-firebase/analytics/package.json
:react-native-firebase_app package.json found at /Users/ken/my-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_analytics:firebase.bom using default value: 33.5.1
:react-native-firebase_analytics package.json found at /Users/ken/my-app/node_modules/@react-native-firebase/analytics/package.json
:react-native-firebase_analytics:version set from package.json: 21.6.0 (21,6,0 - 21006000)
:react-native-firebase_analytics:android.compileSdk using custom value: 35
:react-native-firebase_analytics:android.targetSdk using custom value: 34
:react-native-firebase_analytics:android.minSdk using custom value: 24
:react-native-firebase_analytics:reactNativeAndroidDir /Users/ken/my-app/node_modules/react-native/android

> Configure project :react-native-firebase_app
:react-native-firebase_app package.json found at /Users/ken/my-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_app:firebase.bom using default value: 33.5.1
:react-native-firebase_app:play.play-services-auth using default value: 21.2.0
:react-native-firebase_app package.json found at /Users/ken/my-app/node_modules/@react-native-firebase/app/package.json
:react-native-firebase_app:version set from package.json: 21.6.0 (21,6,0 - 21006000)
:react-native-firebase_app:android.compileSdk using custom value: 35
:react-native-firebase_app:android.targetSdk using custom value: 34
:react-native-firebase_app:android.minSdk using custom value: 24
:react-native-firebase_app:reactNativeAndroidDir /Users/ken/my-app/node_modules/react-native/android

> Configure project :expo

Using expo modules
  - expo-asset (11.0.1)
  - expo-constants (17.0.3)
  - expo-file-system (18.0.4)
  - expo-font (13.0.1)
  - expo-keep-awake (14.0.1)
  - expo-modules-core (2.0.6)


> Configure project :react-native-google-mobile-ads
:react-native-google-mobile-ads package.json found at /Users/ken/my-app/node_modules/react-native-google-mobile-ads/package.json
:react-native-google-mobile-ads package.json found at /Users/ken/my-app/node_modules/react-native-google-mobile-ads/package.json
:react-native-google-mobile-ads:version set from package.json: 14.4.0 (14,4,0 - 14004000)
:react-native-google-mobile-ads:android.compileSdk using custom value: 35
:react-native-google-mobile-ads:android.targetSdk using custom value: 34
:react-native-google-mobile-ads:android.minSdk using custom value: 24
:react-native-google-mobile-ads:reactNativeAndroidDir /Users/ken/my-app/node_modules/react-native/android

> Task :app:processDebugMainManifest FAILED
/Users/ken/my-app/android/app/src/debug/AndroidManifest.xml:32:13-58 Error:
        Attribute property#android.adservices.AD_SERVICES_CONFIG@resource value=(@xml/ga_ad_services_config) from [com.google.android.gms:play-services-measurement-api:22.1.2] AndroidManifest.xml:32:13-58
        is also present at [com.google.android.gms:play-services-ads-lite:23.4.0] AndroidManifest.xml:109:13-59 value=(@xml/gma_ad_services_config).
        Suggestion: add 'tools:replace="android:resource"' to <property> element at AndroidManifest.xml to override.

See https://developer.android.com/r/studio-ui/build/manifest-merger for more information about the manifest merger.


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugMainManifest'.
> Manifest merger failed : Attribute property#android.adservices.AD_SERVICES_CONFIG@resource value=(@xml/ga_ad_services_config) from [com.google.android.gms:play-services-measurement-api:22.1.2] AndroidManifest.xml:32:13-58
        is also present at [com.google.android.gms:play-services-ads-lite:23.4.0] AndroidManifest.xml:109:13-59 value=(@xml/gma_ad_services_config).
        Suggestion: add 'tools:replace="android:resource"' to <property> element at AndroidManifest.xml to override.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s
89 actionable tasks: 5 executed, 84 up-to-date
Error: /Users/ken/my-app/android/gradlew app:assembleDebug -x lint -x test --configure-on-demand --build-cache -PreactNativeDevServerPort=8081 -PreactNativeArchitectures=arm64-v8a exited with non-zero code: 1
Error: /Users/ken/my-app/android/gradlew app:assembleDebug -x lint -x test --configure-on-demand --build-cache -PreactNativeDevServerPort=8081 -PreactNativeArchitectures=arm64-v8a exited with non-zero code: 1
    at ChildProcess.completionListener (/Users/ken/my-app/node_modules/@expo/spawn-async/src/spawnAsync.ts:67:13)
    at Object.onceWrapper (node:events:633:26)
    at ChildProcess.emit (node:events:518:28)
    at maybeClose (node:internal/child_process:1105:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:305:5)
    ...
    at spawnAsync (/Users/ken/my-app/node_modules/@expo/spawn-async/src/spawnAsync.ts:28:21)
    at spawnGradleAsync (/Users/ken/my-app/node_modules/@expo/cli/src/start/platforms/android/gradle.ts:134:28)
    at assembleAsync (/Users/ken/my-app/node_modules/@expo/cli/src/start/platforms/android/gradle.ts:83:16)
    at runAndroidAsync (/Users/ken/my-app/node_modules/@expo/cli/src/run/android/runAndroidAsync.ts:48:24)
```

### Temporary Fix
To fix this issue, manually edit AndroidManifest.xml in the android/app/src/main/ directory. Add tools:replace="android:resource" to the <property> element as follows:

```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
          xmlns:tools="http://schemas.android.com/tools">
    <application>
        <property
            android:name="android.adservices.AD_SERVICES_CONFIG"
            android:resource="@xml/gma_ad_services_config"
            tools:replace="android:resource"/>
    </application>
</manifest>
```


### Platforms

Only on Android

### React Native Info

```Shell
expo-env-info 1.2.1 environment info:
    System:
      OS: macOS 15.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.11.0 - ~/.nodebrew/current/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 10.2.4 - ~/.nodebrew/current/bin/npm
      Watchman: 2024.10.28.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /Users/ken/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.1, iOS 18.1, macOS 15.1, tvOS 18.1, visionOS 2.1, watchOS 11.1
      Android SDK:
        API Levels: 27, 28, 29, 30, 31, 32, 33, 34, 35
        Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.2, 30.0.3, 31.0.0, 33.0.0, 33.0.0, 33.0.1, 33.0.2, 34.0.0, 35.0.0
        System Images: android-30 | Android TV Intel x86 Atom, android-30 | Google Play Intel x86 Atom, android-31 | Android TV Intel x86 Atom, android-35 | Google Play ARM 64 v8a
    IDEs:
      Android Studio: 2024.1 AI-241.18034.62.2412.12266719
      Xcode: 16.1/16B40 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~52.0.11 => 52.0.11 
      react: 18.3.1 => 18.3.1 
      react-native: 0.76.3 => 0.76.3 
    Expo Workflow: bare
```


### Are your using Typescript?

- [X] My project is using Typescript

### package.json

```JSON
{
  "name": "my-app",
  "license": "0BSD",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "start": "expo start --dev-client",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "web": "expo start --web"
  },
  "dependencies": {
    "@react-native-firebase/analytics": "^21.6.0",
    "@react-native-firebase/app": "^21.6.0",
    "expo": "~52.0.11",
    "expo-build-properties": "~0.13.1",
    "expo-status-bar": "~2.0.0",
    "react": "18.3.1",
    "react-native": "0.76.3",
    "react-native-google-mobile-ads": "^14.4.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0"
  },
  "private": true
}
```


### app.json

```JSON
{
  "expo": {
    "name": "my-app",
    "slug": "my-app",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "cover",
      "backgroundColor": "#ffffff"
    },
    "version": "1.0.0",
    "plugins": [
      "@react-native-firebase/app",
      [
        "expo-build-properties",
        {
          "ios": {
            "useFrameworks": "static"
          }
        }
      ]
    ],
    "ios": {
      "bundleIdentifier": "myapp",
      "googleServicesFile": "./GoogleService-Info.plist"
    },
    "android": {
      "googleServicesFile": "./google-services.json",
      "package": "test.myapp"
    }
  },
  "react-native-google-mobile-ads": {
    "android_app_id": "ca-app-pub-****",
    "ios_app_id": "ca-app-pub-****"
  }
}
```


### ios/Podfile

_No response_

### android/build.gradle

_No response_

### android/app/build.gradle

_No response_

### android/settings.gradle

_No response_

### AndroidManifest.xml

_No response_

--- a/android/src/main/AndroidManifest.xml
+++ b/android/src/main/AndroidManifest.xml
@@ -20,6 +20,23 @@
       <meta-data
         android:name="com.google.android.gms.ads.flag.OPTIMIZE_AD_LOADING"
         android:value="${appJSONGoogleMobileAdsOptimizeAdLoading}"/>
+
+      <!-- This may generate a warning during your build:
+
+           > property#android.adservices.AD_SERVICES_CONFIG@android:resource
+           > was tagged at AndroidManifest.xml:23 to replace other declarations
+           > but no other declaration present
+
+           You may safely ignore this warning.
+
+           We must include this in case you also use Firebase Analytics in some
+           of its configurations, as it may also include this file, and the two
+           will collide and cause a build error if we don't set this one to take
+           priority via replacement.
+
+           https://github.com/invertase/react-native-google-mobile-ads/issues/657
+
+      -->
       <property
         android:name="android.adservices.AD_SERVICES_CONFIG"
         android:resource="@xml/gma_ad_services_config"

--- a/android/src/main/AndroidManifest.xml
+++ b/android/src/main/AndroidManifest.xml
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
     package="io.invertase.googlemobileads">
 
     <uses-permission android:name="android.permission.INTERNET" />
@@ -19,5 +20,9 @@
       <meta-data
         android:name="com.google.android.gms.ads.flag.OPTIMIZE_AD_LOADING"
         android:value="${appJSONGoogleMobileAdsOptimizeAdLoading}"/>
+      <property
+        android:name="android.adservices.AD_SERVICES_CONFIG"
+        android:resource="@xml/gma_ad_services_config"
+        tools:replace="android:resource" />
     </application>
 </manifest>

